mbuild/bond_graph.py

```diff
@@ -51,12 +53,12 @@ class BondGraph(object):
     """
 
     def __init__(self):
-        self._adj = defaultdict(set)
+        self._adj = defaultdict(OrderedSet)
 
     def add_node(self, node):
         """Add a node to the bond graph."""
         if not self.has_node(node):
-            self._adj[node] = set()
+            self._adj[node] = OrderedSet()
```

Comment:
Because we use a defaultdict, I don't think it is necessary to initialize each node with an empty set (that's what default dict already does.) Also this function doesn't seem to be used anywhere.

Reply:
So would you recommend removing this function? It seems to me that it might still be useful in some case where you wanted to add to your compound not through adding to the compound object but to the bond_graph. But it does seem a bit convoluted of a use case that maybe it's better to not support it.

We do use remove_node in `compound.py`.

Reply:
yeah, it's probably fine as is. Since `_adj` is a private attribute we wouldn't access it directly anyway. (i.e., it makes sense to have the function even though it's a little weird.)